grails-app/services/de/iteratec/osm/linechart/LineChartTimeSeriesService.groovy

```diff
@@ -0,0 +1,92 @@
+package de.iteratec.osm.linechart
+
+import de.iteratec.osm.csi.Page
+import de.iteratec.osm.measurement.schedule.JobGroup
+import de.iteratec.osm.result.CachedView
+import de.iteratec.osm.result.MeasuredEvent
+import de.iteratec.osm.result.SelectedMeasurand
+import de.iteratec.osm.result.dao.EventResultProjection
+import de.iteratec.osm.result.dao.EventResultQueryBuilder
+import de.iteratec.osm.util.PerformanceLoggingService
+import grails.gorm.transactions.Transactional
+
+import static de.iteratec.osm.util.PerformanceLoggingService.LogLevel.DEBUG
+
+@Transactional
+class LineChartTimeSeriesService {
+
+    PerformanceLoggingService performanceLoggingService
+
+    TimeSeriesChartDTO getTimeSeriesFor(GetLinechartCommand cmd) {
+        Date from = cmd.from.toDate()
+        Date to = cmd.to.toDate()
+        List<JobGroup> jobGroups = null
+        if (cmd.jobGroups) {
+            jobGroups = JobGroup.findAllByIdInList(cmd.jobGroups)
+        }
+        List<Page> pages = null
+        if (cmd.pages) {
+            pages = Page.findAllByIdInList(cmd.pages)
+        }
+        List<MeasuredEvent> measuredEvents = null
+        if (cmd.measuredEvents) {
+            measuredEvents = MeasuredEvent.findAllByIdInList(cmd.measuredEvents)
+        }
+        List<SelectedMeasurand> measurands = cmd.measurands.collect {
+            new SelectedMeasurand(it, CachedView.UNCACHED)
+        }
+        List<EventResultProjection> eventResultProjections
```

Comment:
The logging methods of PerformanceLoggingService return the result of closure to measure (a patch Stefan introduced a while ago). So you may not instantiate the variables needed after performance measurement upfront. e.g.
```suggestion
List<EventResultProjection> eventResultProjections = performanceLoggingService.logExecutionTime(PerformanceLoggingService.LogLevel.DEBUG, 'getting event-results - with builder', 1) { ... }
```
And you don't have to nest blocks to measure.